src/ui/wxWidgets/addeditpropsheet.cpp

```diff
@@ -1584,6 +1614,109 @@ void AddEditPropSheet::OnUseHexCBClick( wxCommandEvent& /* evt */ )
  }
 }
 
+/*!
+ * wxEVT_UPDATE_UI event handler for all command ids
+ */
+
+void AddEditPropSheet::OnUpdateUI(wxUpdateUIEvent& event)
+{
+  bool dbIsReadOnly = m_core.IsReadOnly();
+
+  switch (event.GetId()) {
+  /////////////////////////////////////////////////////////////////////////////
+  // Tab: "Basic"
+  /////////////////////////////////////////////////////////////////////////////
+    case ID_COMBOBOX1:
+    case ID_BUTTON3:
+      event.Enable(!dbIsReadOnly);
+      break;
+    case ID_TEXTCTRL5:
+    case ID_TEXTCTRL1:
+    case ID_TEXTCTRL2:
+    case ID_TEXTCTRL3:
+    case ID_TEXTCTRL4:
+    case ID_TEXTCTRL20:
+    case ID_TEXTCTRL7:
+    {
+      auto window = m_BasicPanel->FindWindow(event.GetId());
+      if (window != nullptr) {
+        auto control = wxDynamicCast(window, wxTextCtrl);
+        if (control != nullptr) {
+          control->SetEditable(!dbIsReadOnly);
+        }
+      }
+      break;
+    }
+
+  /////////////////////////////////////////////////////////////////////////////
+  // Tab: "Additional"
+  /////////////////////////////////////////////////////////////////////////////
+    case ID_TEXTCTRL6:
+    case ID_TEXTCTRL8:
+    {
+      auto window = m_AdditionalPanel->FindWindow(event.GetId());
+      if (window != nullptr) {
+        auto control = wxDynamicCast(window, wxTextCtrl);
+        if (control != nullptr) {
+          control->SetEditable(!dbIsReadOnly);
+        }
+      }
+      break;
+    }
+    case ID_COMBOBOX:
+    case ID_COMBOBOX2:
+    case ID_CHECKBOX1:
+    case ID_SPINCTRL:
+    case ID_GRID:
+    case ID_BUTTON1:
+    case ID_BUTTON4:
+      event.Enable(!dbIsReadOnly);
```

Comment:
This looks cryptic. Maybe these controls didn't have any logic attached to them before, so they had these generic names, but now they have different behavior, so maybe it's time to give them meaningful names?

Reply:
Awesome!